myokit/formats/diffsl/_exporter.py

```diff
@@ -0,0 +1,523 @@
+        """
+        Prepare the model for export to DiffSL.
+        """
+        # Rewrite model so that any Markov models have a 1-sum(...) state
+        # This also clones the model, so that changes can be made
+        model = markov.convert_markov_models_to_compact_form(model)
+
+        # Remove all model bindings apart from time and pace
+        _ = myokit._prepare_bindings(
+            model,
+            {
+                'time': 't',
+                'pace': 'pace',
+            },
+        )
+
+        # Remove hardcoded stimulus protocol, if any
+        guess.remove_embedded_protocol(model)
+
+        # Get / try to guess some model variables
+        time = model.time()  # engine.time
+        vm = self._guess_membrane_potential(model)  # Vm
+        cm = guess.membrane_capacitance(model)  # Cm
+        currents = self._guess_currents(model)
+
+        time_factor = time_factor_inv = None
+        if convert_units:
+            # Convert currents to A/F
+            helpers = [] if cm is None else [cm.rhs()]
+            for var in currents:
+                self._convert_current_unit(var, helpers=helpers)
+
+            # Convert potentials to mV
+            self._convert_potential_unit(vm)
+
+            time_factor, time_factor_inv = self._get_time_factor(time)
+
+        # Find HH state variables and their alphas, betas, and taus
+        alphas = set()
+        betas = set()
+        taus = set()
+
+        for var in model.states():
+            inf_tau = hh.get_inf_and_tau(var, vm)
+            if inf_tau is not None:
+                taus.add(inf_tau[1])
+                continue
+
+            alpha_beta = hh.get_alpha_and_beta(var, vm)
+            if alpha_beta is not None:
+                alphas.add(alpha_beta[0])
+                betas.add(alpha_beta[1])
+                continue
+
+        # Add time conversion factor for states and HH variables
+        if time_factor is not None:
+            for var in model.states():
+                var.set_rhs(myokit.Multiply(var.rhs(), time_factor_inv))
+            for var in alphas:
+                var.set_rhs(myokit.Multiply(var.rhs(), time_factor_inv))
+            for var in betas:
+                var.set_rhs(myokit.Multiply(var.rhs(), time_factor_inv))
+            for var in taus:
+                var.set_rhs(myokit.Multiply(var.rhs(), time_factor))
+
+        # Add intermediary variables on rhs of state derivatives
+        self._prep_derivatives(model)
+
+        return model, currents
+
+    def _generate_diffsl(self, model, extra_out_vars=None):
+        """
+        Generate a DiffSL model from a prepped Myokit model.
+        DiffSL outputs will be set to state variables and extra_out_vars
+        """
+
+        # Create DiffSL-compatible variable names
+        var_to_name = self._create_diffsl_variable_names(model)
+
+        # Create a naming function
+        def var_name(e):
+            if isinstance(e, myokit.LhsExpression):
```

Comment:
Is this safe because all `LhsExpressions` here are guaranteed to be `Name` and not `Derivative` (filtered out by intermediary variable adding code in prepare)?
Might be nicer to explicitly write `isinstance(e, myokit.Name)` so that it will fail properly if that ever changes

Reply:
Yes, that's why it's safe. Thanks, I'll use `myokit.Name` to narrow it down.

Reply:
Added naming for `myokit.Derivative` in 1a21249